app/models/time_profile.rb

```diff
@@ -37,31 +38,31 @@ def global?
     profile_type == "global"
   end
 
-  def ts_in_profile?(ts, default_tz = DEFAULT_TZ)
-    ts = Time.parse(ts) if ts.kind_of?(String)
-    self.ts_day_in_profile?(ts, default_tz) && self.ts_hour_in_profile?(ts, default_tz)
+  def ts_in_profile?(tstamp, default_tz = DEFAULT_TZ)
+    tstamp = Time.zone.parse(tstamp) if tstamp.kind_of?(String)
+    ts_day_in_profile?(tstamp, default_tz) && ts_hour_in_profile?(tstamp, default_tz)
   end
 
-  def ts_hour_in_profile?(ts, default_tz = DEFAULT_TZ)
-    ts = Time.parse(ts) if ts.kind_of?(String)
-    hours.include?(ts.in_time_zone(tz_or_default(default_tz)).hour)
+  def ts_hour_in_profile?(tstamp, default_tz = DEFAULT_TZ)
+    tstamp = Time.zone.parse(tstamp) if tstamp.kind_of?(String)
+    hours.include?(tstamp.in_time_zone(tz_or_default(default_tz)).hour)
   end
 
-  def ts_day_in_profile?(ts, default_tz = DEFAULT_TZ)
-    ts = Time.parse(ts) if ts.kind_of?(String)
-    days.include?(ts.in_time_zone(tz_or_default(default_tz)).wday)
+  def ts_day_in_profile?(tstamp, default_tz = DEFAULT_TZ)
+    tstamp = Time.zone.parse(tstamp) if tstamp.kind_of?(String)
+    days.include?(tstamp.in_time_zone(tz_or_default(default_tz)).wday)
   end
 
   def profile
-    read_attribute(:profile) || (self.profile = {})
+    self[:profile] || (self.profile = {})
```

Comment:
Technically this should be

```suggestion
    super || (self.profile = {})
```

Reply:
Updated.